cluster/charts/rook-ceph/templates/resources.yaml

```diff
@@ -12,19 +12,24 @@ spec:
     singular: cephcluster
   scope: Namespaced
   version: v1
+  preserveUnknownFields: false
```

Comment:
Enabling this more strict mode could potentially break upgraded clusters. For backward compatibility, we have some settings that are still supported in the code, but the schema has really only been used for validation of new clusters. 

For example, the setting `hostNetwork: true` is now replaced by `provider: host`. Upgraded clusters would expect to continue using host networking, which is working today. But since `hostNetwork` isn't in the latest schema, they would be broken unless we researched all the old settings and added them back to the schema.

Reply:
I think your example can be handled by using validation fields like `deprecated`, https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#validation. 

But I agree, overall, we would have to be really careful and one other problem is that because our Kubernetes version support range is so wide, we cannot take advantage of all the open API validation, which is a bummer...

Reply:
Yeap.. That's why i just open an issue with suggesting feature, not bug report. If crd use pruning unknown fields, cr could become less flexible as you commented.

If rook chooses to enable pruning unknown fields, in the example of @travisn, the cluster.yaml with `hostNetwork:true` is failed to apply by validation check. So I think the user who wants to upgrade cluster will notice crd was changed, then the user should apply the new crd first, before upgrading cluster.

I also cannot think any flexible idea to use pruning and to have backward compatiblity of crd...